spec/integration/data_stream/fields/fields.spec.yml

```
@@ -548,12 +557,26 @@ spec:
               enum:
                 - group
                 - nested
+      - if:
+          required:
+            - subobjects
+        then:
+          properties:
+            type:
+              enum:
+                - object
+          required:
+            - type            
 
     required:
     - name
 
 # JSON patches for newer versions should be placed on top
 versions:
+  - before: 3.1.0
+    patch:
+      - op: remove
+        path: "/items/allOf/8" # removing subobjects when type is object
```

Remove `subobjects` property too?
```suggestion
        path: "/items/allOf/8" # removing subobjects when type is object
      - op: remove
        path: "/items/properties/subobjects"
```